import { HttpException } from "@nestjs/common"

export enum errors {
  ACCOUNT_NOT_FOUND=901,
  AUTH_FAILED,
  BIN_INCORRECT,
  BIN_REQUIRED,
  CAPTCHA_INCORRECT,
  CAPTCHA_REQUIRED,  
  CONNECTION_ERROR,
  FORBIDDEN,
  IIN_INCORRECT,
  IIN_REQUIRED,
  INSUFFICIENT_FUNDS,
  PASSWORD_MINLENGTH,
  PASSWORD_REQUIRED,
  PHONE_EXISTS,
  PHONE_INCORRECT,
  PHONE_REQUIRED,
  PROFILEFIELD_INCORRECT,
  SHOP_DELETED,
  SHOP_HIDDEN,
  SHOP_NOT_FOUND,
  SMSCODE_INCORRECT,
  SMSCODE_NOT_FOUND,
  SMSCODE_TRY_LATER,
  SMSCODEORIGIN_INCORRECT,
  USER_BLOCKED,
  USER_DELETED,
  USERTYPE_INCORRECT,
  USERNAME_REQUIRED,
}

export const myException = (code: number): void => {
  throw new HttpException(errors[code], code)
} 
